lib/std/builtin.zig

```diff
@@ -478,6 +487,26 @@ pub const CallingConvention = union(enum(u8)) {
         };
     };
 
+    /// Options for the spirv calling conventions.
+    pub const SpirvOptions = struct {
+        /// The execution mode.
+        mode: ?ExecutionMode = null,
+
+        pub const ExecutionMode = union(enum) {
+            pixel_center_integer,
+            origin_upper_left,
+            origin_lower_left,
+            early_fragment_tests,
+            depth_replacing,
+            depth_greater,
+            depth_less,
+            depth_unchanged,
+            local_size: LocalSize,
+        };
+    };
```

Comment:
Do these all in fact affect the calling sequence / ABI?

Also do they really apply to all SPIR-V CCs?

Reply:
Well, functions with those calling conventions are only meant to be program entry-points and those options affect how they are executed.

No, `local_size` is only for `spirv_kernel` and the rest is for `spirv_fragment`. Though i think it's good to be future proof in this case. I also got no arguments against dividing them btw.

Reply:
What I'm getting at is that a builtin function that sets the execution mode of the current function might be a better fit, at least if the mode isn't observable for the host that calls the function. I don't know if that's actually the case.

Reply:
Hmm that isn't any better than the status quo (calling `std.gpu` functions). the main reason for adding these options is that fragment and kernel shaders requires an execution mode. though i think we can go with your suggestion to set a default one and add the ability to override it with a builtin function.